### ISSUE TITLE
[bug] False positive in `@eslint-react/naming-convention/use-state` fo state with no setters

### ISSUE THREAD

**Issue by kachkaev** [Rel1cx/eslint-react#1352]
### Describe the bug

I would like to do something like:

```ts
const [memoisedValue] = React.useState(() => calculateValue());
```

In this case `useState` is preferred over `useMemo` because `useMemo` does not guarantee that `calculateValue` is only called once.

However, the above code produces this ESLint warning:

```
useState should be destructured into a value and setter pair, e.g., const [state, setState] = useState(...).
eslint[@eslint-react/naming-convention/use-state](https://eslint-react.xyz/docs/rules/naming-convention-use-state)
```

### Expected behavior

Pattern like `const [state] = useState(getInitialValue)` is not flagged. Setter is optional.

### Platform and versions

```Text
"@eslint-react/eslint-plugin": "2.3.13"
"eslint": "9.39.2",
```

### Stack trace

```Text

```
